module/filter/doubleFilter.py

```diff
@@ -10,33 +10,45 @@
                      by Bastian Schroll
 
 @file:        doubleFilter.py
-@date:        15.01.2018
-@author:      Bastian Schroll
-@description: Class to implement a filter for double alarms
-@todo test, refactor and document / check_msg is not implemented yet
+@date:        09.07.2020
+@author:      Bastian Schroll, b-watch
+@description: Filter module for double packages
 """
 import logging
+from module.moduleBase import ModuleBase
+
+# ###################### #
+# Custom plugin includes #
 import time
+# ######################
 
 logging.debug("- %s loaded", __name__)
 
 
-class DoubleFilter:
-    """!Double Filter Class"""
-
+class BoswatchModule(ModuleBase):
+    """!Description of the Module"""
     def __init__(self, config):
-        """!init"""
-        self._config = config
+        """!Do not change anything here!"""
+        super().__init__(__name__, config)  # you can access the config class on 'self.config'
         self._filterLists = {}
+        logging.debug("Configured ignoreTime: %d", self.config.get("ignoreTime", default=10))
+        logging.debug("Configured maxEntry: %d", self.config.get("maxEntry", default=10))
 
-    def filter(self, bwPacket):
+    def onLoad(self):
+        """!Called by import of the plugin
+        Remove if not implemented"""
+        pass
 
+    def doWork(self, bwPacket):
+        """!start an run of the module.
+
+        @param bwPacket: A BOSWatch packet instance"""
         if bwPacket.get("mode") == "fms":
-            scanWord = "fms"
+            filterFields = ["fms"]
         elif bwPacket.get("mode") == "pocsag":
-            scanWord = "ric"
+            filterFields = self.config.get("pocsagFields", default=["ric"])
```

Comment:
Ich weiss nicht, ob es vielleicht noch sinn macht hier als default noch die subric mit aufzunehmen, nur ne Idee
```suggestion
            filterFields = self.config.get("pocsagFields", default=["ric", "subric"])
```

Reply:
@b-watch Was meinst du? ich würde mich an dem Punkt raus halten, da wir bei uns kein POCSAG haben.
Falls ja, bitte die Doku nicht vergessen anzupassen.

Reply:
macht Sinn, auch bei uns sind die subrics relevant. Hab es angepasst.